Framework/Crystal/test/IntegratePeakTimeSlicesTest.h

```diff
@@ -237,18 +233,16 @@ class IntegratePeakTimeSlicesTest : public CxxTest::TestSuite {
   /**
    *   Calculates Q
    */
-  double calcQ(RectangularDetector_const_sptr bankP,
-               boost::shared_ptr<const Instrument> instPtr, int row, int col,
+  double calcQ(const DetectorInfo &detectorInfo, const detid_t detID,
                double time) {
-    boost::shared_ptr<Detector> detP = bankP->getAtXY(col, row);
-
-    double L2 = detP->getDistance(*(instPtr->getSample()));
+    const auto wsIndex = detectorInfo.indexOf(detID);
+    const auto L2 = detectorInfo.l2(wsIndex);
 
     Kernel::Units::MomentumTransfer Q;
     std::vector<double> x;
     x.push_back(time);
-    double L1 = instPtr->getSample()->getDistance(*(instPtr->getSource()));
-    Kernel::V3D pos = detP->getPos();
+    const auto L1 = detectorInfo.l1();
+    const auto pos = detectorInfo.position(wsIndex);
     double ScatAng = fabs(asin(pos.Z() / pos.norm()));
```

Comment:
Not easily. They refer to the detector by its row and column in a detector bank. I could pass DetectorInfo to the method and get the L1 from it, though, but is it really worth it?